src/NzbDrone.Common/Disk/DiskProviderBase.cs

```diff
@@ -189,6 +189,17 @@ public long GetFileSize(string path)
             }
 
             var fi = new FileInfo(path);
+
+            // If the file is a symlink, resolve the target path and get the size of the target file.
+            if ((fi.Attributes & FileAttributes.ReparsePoint) == FileAttributes.ReparsePoint)
```

Comment:
This can be simplified

```suggestion
            if (fi.Attributes.HasFlag(FileAttributes.ReparsePoint))
```